crypto/fipsmodule/ec/make_tables.go

```diff
@@ -341,6 +412,35 @@ func bigIntToU64s(curve elliptic.Curve, n *big.Int) []uint64 {
 	return ret
 }
 
+// Convert big int to an array of 58-bit digits.
+// This is needed for P-521 Fiat-crypto implementation.
+func bigIntToU58s(curve elliptic.Curve, n *big.Int) []uint64 {
+	words := (curve.Params().BitSize + 57) / 58
+	ret := make([]uint64, words)
+	mask := big.NewInt((1 << 58) - 1)
+	tmp := new(big.Int).Set(n)
+	for i := 0; i < words; i++ {
+		ret[i] = new(big.Int).And(tmp, mask).Uint64()
+		tmp.Rsh(tmp, 58)
+	}
+	return ret
+}
+
+// Convert big int to an array of digits where each digit
+// has bit-size as specified in the input bitSizes array
+// This is needed for P-521 Fiat-crypto implementation.
```

Comment:
```suggestion
// This is needed for P-521 Fiat-crypto implementation in fiat/p521_32.h
```

Reply:
If you take the suggestion, I think it would be good to put `third_party` in the path, sorry.